src/clj/commiteth/middleware.clj

```diff
@@ -64,15 +69,22 @@
         (wrap-authentication backend)
         (wrap-authorization backend))))
 
+(defstate session-cleaner
+  :start #(start-cleaner *db*)
+  :stop  #(stop-cleaner session-cleaner))
+
+(def redirect-to-home-page (response/redirect "/"))
+
 (defn wrap-base [handler]
   (-> ((:middleware defaults) handler)
       wrap-auth
       wrap-flash
-      (wrap-session {:timeout      (* 60 60 6)
-                     :cookie-attrs {:http-only true}})
+      (wrap-idle-session-timeout {:timeout-response redirect-to-home-page
+                                  :timeout          (* 60 60 6)})
       (wrap-defaults
        (-> site-defaults
            (assoc-in [:security :anti-forgery] false)
-           (dissoc :session)))
+           (assoc-in [:session] {:cookie-attrs {:http-only true}
+                                 :store        (jdbc-store *db*)})))
```

Comment:
Might be missing something here but could we use `ring.middleware.session` for this?

```clj
(ring.middleware.session/wrap-session
   {:cookie-attrs {:http-only true}
    :store        (jdbc-store *db*)})
```